dig_test.go

```diff
@@ -1642,6 +1642,34 @@ func TestProvideCycleFails(t *testing.T) {
 		)
 	})
 
+	t.Run("parameters only on dry run", func(t *testing.T) {
```

Comment:
So, as it is, this test seems only to be confirming that a Dry container will error on introducing a cycle, but isn't actively asserting that the functions aren't invoked.

A naïve soltuion might be something like:
```
var invocations int

newA := func(*C) *A {
    invocations ++
    return &A{}
}
```

Reply:
It's a good idea, but unfortunately in these tests, since we are asserting for an error, we don't get to part where things are invoked, so even if we aren't in Dry mode, we will never call these functions, so invocations will always be 0.

I should create a test case where Dry is true and we don't error, and we can use your suggestion.

Reply:
Added a test when a panic within Provide does not run (or test would fail).